packages/common/src/components/MapSearch.tsx

```diff
@@ -0,0 +1,206 @@
+import React, { forwardRef, useEffect, useRef, useState } from "react";
+import { GeolocationControl, Map, Placemark, SearchControl } from "react-yandex-maps";
+import { Button } from "antd";
+
+type Adress = {
+  address: string;
+  coordinates: number[];
+};
+
+type MapSearchProps = {
+  setAddressAntd: string;
+  handleOpen: () => true;
+  setSearchData: (searchData: Adress) => void;
+  handleCancel: () => void;
+  onAddNewLocation: () => void;
+  searchdata: Adress;
+};
+
+export const MapSearch = forwardRef((Props: React.PropsWithChildren<MapSearchProps>, ref) => {
+  const { setAddressAntd, handleCancel, onAddNewLocation, handleOpen, setSearchData, searchdata } =
+    Props;
+
+  const searchControlRef = useRef<ymaps.control.SearchControl | null>;
+
+  const [mapConstructor, setMapConstructor] = useState(null);
+
+  const [addressCoord, setAddressCoordMap] = useState<number[]>();
+
+  const [hasFetched, setHasFetched] = useState<boolean>(true); // Состояние для отслеживания вызова функции
+
+  const mapOptions = {
+    modules: ["geocode", "SuggestView"],
+    defaultOptions: { suppressMapOpenBlock: true },
+    width: "auto",
+    height: 300,
+  };
+  const mapState = {
+    center: [55.76, 37.64],
+    zoom: 12,
+    controls: [],
+  };
+
+  const geolocationOptions = {
+    defaultOptions: { maxWidth: 128 },
+    defaultData: { content: "Determine" },
+  };
+
+  useEffect(() => {
+    const fetchAddressCoordStateMap = async () => {
+      if (!mapConstructor) return;
+      try {
+        // @ts-ignore
+        const cords = await mapConstructor.geocode(setAddressAntd);
```

Comment:
It's not a joke. 
`setAddressAntd` - this naming for a string is really frustrating!